src/Build.UnitTests/Scanner_Tests.cs

```diff
@@ -104,16 +101,31 @@ public void IllFormedProperty()
         /// <summary>
         /// Tests the space errors case
         /// </summary>
-        [Fact]
-        public void SpaceProperty()
+        [Theory]
+        [InlineData("$(x )")]
+        [InlineData("$( x)")]
+        [InlineData("$([MSBuild]::DoSomething($(space ))")]
+        [InlineData("$([MSBuild]::DoSomething($(_space ))")]
+        public void SpaceProperty(string pattern)
         {
-            Scanner lexer = new Scanner("$(x )", ParserOptions.AllowProperties);
+            Scanner lexer = new Scanner(pattern, ParserOptions.AllowProperties);
             AdvanceToScannerError(lexer);
             Assert.Equal("IllFormedPropertySpaceInCondition", lexer.GetErrorResource());
+        }
 
-            lexer = new Scanner("$( x)", ParserOptions.AllowProperties);
+        /// <summary>
+        /// Tests the space not next to end so no errors case
+        /// </summary>
+        [Theory]
+        [InlineData("$(x.StartsWith( 'y' ))")]
+        [InlineData("$(x.StartsWith ('y'))")]
```

Comment:
```suggestion
        [InlineData("$(x.StartsWith ('y'))")]
        [InlineData("$(x.StartsWith('$('))")]
```

Reply:
The second case doesn't have any spaces, so it would pass this check trivially. Would it be more interesting to add a space?

Reply:
It looks like we already fail the `$(` case because the condition expander can't handle that as a literal, so adding it wouldn't be helpful.